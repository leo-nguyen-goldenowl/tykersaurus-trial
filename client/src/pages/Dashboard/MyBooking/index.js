import React from 'react'

const MyBooking = () => {
  return (
    <div className='my-booking'>
      <table>
        <thead>
          <tr>
            <th>Date</th>
            <th>Tee time</th>
            <th>Quantity of Players</th>
            <th>Start Tee</th>
            <th>Golf Course</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>29 December 2020</td>
            <td>17:43</td>
            <td>4</td>
            <td>
              1<sup>st</sup>
            </td>
            <td>Classic course</td>
          </tr>
        </tbody>
      </table>
    </div>
  )
}

export default MyBooking
