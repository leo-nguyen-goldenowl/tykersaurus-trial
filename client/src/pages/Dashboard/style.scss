@import "src/assets/scss/_variables.scss";
@import "src/assets/scss/_mixins.scss";
@import "src/assets/scss/_extends.scss";

.dashboard-page {
  background: $snow;
  min-height: 100vh;
  &__book-course {
    background-image: url("../../assets/images/intro--large.png");
    background-position: center;
    background-size: cover;
    position: relative;
    height: 600px;
    .lazy-load-image-background {
      height: 100%;
    }
    &__background {
      width: 100%;
      height: 100%;
    }
    &__form {
      width: calc(100% - 190px);
      position: absolute;
      bottom: 0;
      left: 50%;
      transform: translate(-50%, 50%);
      display: flex;
      justify-content: center;
    }
  }
}
