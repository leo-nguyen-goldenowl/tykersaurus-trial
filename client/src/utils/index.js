export const getValueFromStorage = (name) => {
  return localStorage.getItem(name)
}
