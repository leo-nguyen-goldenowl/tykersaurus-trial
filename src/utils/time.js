const moment = require('moment')

const convertTeeTimeToMinute = (teeTime) => {
  const listPartOfTeeTime = teeTime.split(':')
  const hourTeeTime = listPartOfTeeTime[0]
  const minuteTeeTime = listPartOfTeeTime[1]

  return Number(hourTeeTime * 60 + minuteTeeTime)
}

const getMaxDate = () => {
  const todayMoment = moment()
  return moment(todayMoment).add(
    moment(moment()).diff(
      moment(
        moment().add(-1, 'days').format('MM/DD/YYYY 23:00'),
        'MM-DD-YYYY hh:mm'
      ),
      'seconds'
    ) > 0
      ? 7
      : 6,
    'days'
  )
}

const accessBooking = (date) => {
  const maxDate = getMaxDate()

  return (
    moment(moment(maxDate).format('MM/DD/YYYY 00:00')).diff(
      moment(date),
      'seconds'
    ) >= 0
  )
}

module.exports = { convertTeeTimeToMinute, accessBooking, getMaxDate }
