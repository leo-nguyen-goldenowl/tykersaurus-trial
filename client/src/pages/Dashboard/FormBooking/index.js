import React from 'react'
import moment from 'moment'
import { useMergeState } from 'hooks'

import ButtonDefault from 'components/Button/Default'
import SelectMini from 'components/Select/Mini'
import SelectLarge from 'components/Select/Large'
import DatePicker from 'components/Picker/Date'
import TimePicker from 'components/Picker/Time'

import './style.scss'

const FormBooking = () => {
  const listSelectTop = [
    {
      label   : 'Session',
      name    : 'session',
      listItem: [
        { name: 'Morning', value: '07:00,11:59' },
        { name: 'Afternoon', value: '12:00,18:00' }
      ]
    },
    {
      label   : 'Hole',
      name    : 'hole',
      listItem: [
        { name: '9 Holes', value: 9 },
        { name: '18 Holes', value: 18 }
      ]
    },
    {
      label   : 'Quantity of Players',
      name    : 'player',
      listItem: [
        { name: '2 Players', value: 2 },
        { name: '3 Players', value: 3 },
        { name: '4 Players', value: 4 }
      ]
    }
  ]

  const [dataBooking, setDataBooking] = useMergeState({
    session     : '07:00,11:59',
    hole        : 9,
    player      : 2,
    course      : 'Classic Course',
    date        : moment().format('ddd, MMM DD YYYY'),
    teeTimeRange: {
      from: moment(),
      to  : moment()
    }
  })

  const handleChangeDate = (value) => {
    setDataBooking({ ['date']: moment(value).format('ddd, MMM DD YYYY') })
  }

  const handleChangeSelect = (e) => {
    const { name, value } = e.target
    setDataBooking({ [name]: value })
  }

  const handleChangeTime = (value, type) => {
    setDataBooking({
      teeTimeRange: {
        ...dataBooking.teeTimeRange,
        [type]: value
      }
    })
  }

  const handleValidDataBooking = () => {
    const { from: fromTeeTime, to: toTeetime } = dataBooking.teeTimeRange

    return (
      moment(moment(toTeetime)).diff(moment(moment(fromTeeTime)), 'minutes') >=
      0
    )
  }

  const handleSubmit = () => {
    if (handleValidDataBooking()) {
      console.log('Ready to book')
    } else {
      console.log('Invalid data')
    }
  }

  return (
    <div className='form-booking'>
      <div className='form-booking__top'>
        {listSelectTop.map((select, idx) => (
          <div className='form-booking__top__item' key={idx}>
            <SelectMini
              label={select.label}
              name={select.name}
              value={dataBooking[select.name]}
              listItem={select.listItem}
              onChange={handleChangeSelect}
            />
          </div>
        ))}
      </div>
      <div className='form-booking__bottom'>
        <div className='form-booking__bottom__item'>
          <SelectLarge
            label='Course'
            name='course'
            value={dataBooking.course}
            onChange={handleChangeSelect}
            listItem={[
              { name: 'Classic Course', value: 'Classic Course' },
              {
                name : 'Masters Course',
                value: 'Masters Course'
              }
            ]}
          />
        </div>
        <div className='form-booking__bottom__item'>
          <DatePicker
            label='Calendar'
            name='calendar'
            value={dataBooking.date}
            type='calendar'
            onChange={handleChangeDate}
          />
        </div>
        <div className='form-booking__bottom__item'>
          <TimePicker
            value={dataBooking.teeTimeRange}
            label='Time'
            onChange={handleChangeTime}
          />
        </div>
      </div>
      <p className='form-booking__description'>
        <span>Description:</span> Lorem ipsum dolor sit amet, modus inimicus eum
        no <span>7AM</span> at nam modus <span>1 week</span> consul. Mucius
        oportere vel id. Ut solum malorum conclusionemque qui. Mea te graeci
        eruditi honestatis. Ne iisque placerat est, mel in dicunt invenire. Te
        his delenit eloquentiam.
      </p>
      <div className='form-booking__btn-submit'>
        <ButtonDefault type='button' onClick={handleSubmit}>
          Book Now
        </ButtonDefault>
      </div>
    </div>
  )
}

export default FormBooking
