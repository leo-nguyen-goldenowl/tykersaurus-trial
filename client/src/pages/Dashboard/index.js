import React from 'react'

import HeaderPage from 'containers/HeaderPage'
import FormBooking from './FormBooking'
import MyBooking from './MyBooking'

import './style.scss'

const DashboardPage = () => {
  return (
    <div className='dashboard-page'>
      <HeaderPage />
      <div className='dashboard-page__book-course'>
        <div className='dashboard-page__book-course__form'>
          <FormBooking />
        </div>
      </div>
      <MyBooking />
    </div>
  )
}

export default DashboardPage
