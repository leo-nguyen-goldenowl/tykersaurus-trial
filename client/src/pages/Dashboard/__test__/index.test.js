import { testSnapshots } from "utils/test/test.utils";
import Dashboard from "../index";

jest
  .mock("../FormBooking", () => "FormBooking")
  .mock("../MyBooking", () => "MyBooking")
  .mock("containers/HeaderPage", () => "HeaderPage");

describe("Test Dashboard component", () => {
  testSnapshots(Dashboard, [
    {
      props: {},
      description: "Dashboard component should match snapshots",
    },
  ]);
});
