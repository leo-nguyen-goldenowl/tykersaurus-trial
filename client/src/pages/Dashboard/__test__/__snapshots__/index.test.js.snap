// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Dashboard component snapshots Dashboard component should match snapshots 1`] = `
<div
  className="dashboard-page"
>
  <HeaderPage />
  <div
    className="dashboard-page__book-course"
  >
    <div
      className="dashboard-page__book-course__form"
    >
      <FormBooking />
    </div>
  </div>
  <div
    className="dashboard-page__my-booking"
  >
    <div
      className="dashboard-page__my-booking__list"
    >
      <MyBooking />
    </div>
  </div>
</div>
`;
